lib/optimize/CommonsChunkPlugin.js

```diff
@@ -2,183 +2,176 @@
-	if(options.children) this.selectedChunks = false;
-	this.async = options.async;
-	this.minSize = options.minSize;
-	this.ident = __filename + (nextIdent++);
-}
+class CommonsChunkPlugin {
+	constructor(options) {
+		if(arguments.length > 1) {
+			throw new Error(`Deprecation notice: CommonsChunkPlugin now only takes a single argument. Either an options
+object *or* the name of the chunk.
+Example: if your old code looked like this:
+	new webpack.optimize.CommonsChunkPlugin('vendor', 'vendor.bundle.js')
 
-module.exports = CommonsChunkPlugin;
-CommonsChunkPlugin.prototype.apply = function(compiler) {
-	var chunkNames = this.chunkNames;
-	var filenameTemplate = this.filenameTemplate;
-	var minChunks = this.minChunks;
-	var selectedChunks = this.selectedChunks;
-	var asyncOption = this.async;
-	var minSize = this.minSize;
-	var ident = this.ident;
-	compiler.plugin("this-compilation", function(compilation) {
-		compilation.plugin(["optimize-chunks", "optimize-extracted-chunks"], function(chunks) {
-			// only optimize once
-			if(compilation[ident]) return;
-			compilation[ident] = true;
+You would change it to:
+	new webpack.optimize.CommonsChunkPlugin({ name: 'vendor', filename: 'vendor.bundle.js' })
 
-			var commonChunks;
-			if(!chunkNames && (selectedChunks === false || asyncOption)) {
-				commonChunks = chunks;
-			} else if(Array.isArray(chunkNames) || typeof chunkNames === "string") {
-				commonChunks = [].concat(chunkNames).map(function(chunkName) {
-					var chunk = chunks.filter(function(chunk) {
-						return chunk.name === chunkName;
-					})[0];
-					if(!chunk) {
-						chunk = this.addChunk(chunkName);
-					}
-					return chunk;
-				}, this);
-			} else {
-				throw new Error("Invalid chunkNames argument");
-			}
-			commonChunks.forEach(function processCommonChunk(commonChunk, idx) {
-				var usedChunks;
-				if(Array.isArray(selectedChunks)) {
-					usedChunks = chunks.filter(function(chunk) {
-						if(chunk === commonChunk) return false;
-						return selectedChunks.indexOf(chunk.name) >= 0;
-					});
-				} else if(selectedChunks === false || asyncOption) {
-					usedChunks = (commonChunk.chunks || []).filter(function(chunk) {
-						// we can only move modules from this chunk if the "commonChunk" is the only parent
-						return asyncOption || chunk.parents.length === 1;
-					});
+The available options are:
+	name: string
+	names: string[]
+	filename: string
+	minChunks: number
+	chunks: string[]
+	children: boolean
+	async: boolean
+	minSize: number`);
+		}
+
+		if(Array.isArray(options) || typeof options === "string") {
+			options = {
+				name: options
+			};
+		}
+		this.chunkNames = options.name || options.names;
+		this.filenameTemplate = options.filename;
+		this.minChunks = options.minChunks;
+		this.selectedChunks = options.chunks;
+		if(options.children) this.selectedChunks = false;
+		this.async = options.async;
+		this.minSize = options.minSize;
+		this.ident = __filename + (nextIdent++);
```

Comment:
`this.ident` should also not be necessary anymore (unless external code depends on it?)